### ISSUE TITLE
`test_ecco.jl` has an error

### ISSUE THREAD

**Issue by francispoulin** [CliMA/ClimaOcean.jl#109]
When I try running `test_ecco.jl` it has an error, see below for details.

It seems that i need to set ECCO_USERNAME in the environment variables.   I see that in `ecco_metgadata.jl` there are the following instructions to specify my username and password.

```ECCO_USERNAME=myuser ECCO_PASSWORD=mypasswrd julia```

I tried this (but admit it seems insecure to have to show my password) and I still get an error that authentication failed.  I tested the username and password on the NASA website and it worked fine.  Any idea as to what I might be missing?

```
julia> include("test_ecco.jl")
[ Info: Testing ecco_field on CPU...
ECCO fields utilities: Error During Test at /home/fpoulin/Software/ClimaOcean.jl/test/test_ecco.jl:11
  Got exception outside of a @test
  ArgumentError: Could not find the username for https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/interp_monthly/. Please provide a username in the ECCO_USERNAME environment variable.
  Stacktrace:
    [1] download_dataset!(metadata::ECCOMetadata{StepRange{DateTimeProlepticGregorian, Month}, ECCO4Monthly}; url::String)
      @ ClimaOcean.DataWrangling.ECCO ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:156
    [2] download_dataset!
      @ ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:144 [inlined]
    [3] ECCO_field_time_series(metadata::ECCOMetadata{StepRange{DateTimeProlepticGregorian, Month}, ECCO4Monthly}; architecture::CPU, time_indices_in_memory::Int64, time_indexing::Oceananigans.OutputReaders.Cyclical{Nothing}, grid::Nothing)
      @ ClimaOcean.DataWrangling.ECCO ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_restoring.jl:123
    [4] ECCO_field_time_series
      @ ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_restoring.jl:112 [inlined]
    [5] ECCO_restoring_forcing(metadata::ECCOMetadata{StepRange{DateTimeProlepticGregorian, Month}, ECCO4Monthly}; architecture::CPU, time_indices_in_memory::Int64, time_indexing::Oceananigans.OutputReaders.Cyclical{Nothing}, mask::Int64, timescale::Float64, grid::Nothing)
      @ ClimaOcean.DataWrangling.ECCO ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_restoring.jl:271
    [6] ECCO_restoring_forcing
      @ ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_restoring.jl:263 [inlined]
    [7] macro expansion
      @ ~/Software/ClimaOcean.jl/test/test_ecco.jl:21 [inlined]
    [8] macro expansion
      @ ~/Software/julia-1.10.0/share/julia/stdlib/v1.10/Test/src/Test.jl:1577 [inlined]
    [9] top-level scope
      @ ~/Software/ClimaOcean.jl/test/test_ecco.jl:12
   [10] include(fname::String)
      @ Base.MainInclude ./client.jl:489
   [11] top-level scope
      @ REPL[12]:1
   [12] eval
      @ Core ./boot.jl:385 [inlined]
   [13] eval_user_input(ast::Any, backend::REPL.REPLBackend, mod::Module)
      @ REPL ~/Software/julia-1.10.0/share/julia/stdlib/v1.10/REPL/src/REPL.jl:150
   [14] repl_backend_loop(backend::REPL.REPLBackend, get_module::Function)
      @ REPL ~/Software/julia-1.10.0/share/julia/stdlib/v1.10/REPL/src/REPL.jl:246
   [15] start_repl_backend(backend::REPL.REPLBackend, consumer::Any; get_module::Function)
      @ REPL ~/Software/julia-1.10.0/share/julia/stdlib/v1.10/REPL/src/REPL.jl:231
   [16] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool, backend::Any)
      @ REPL ~/Software/julia-1.10.0/share/julia/stdlib/v1.10/REPL/src/REPL.jl:389
   [17] run_repl(repl::REPL.AbstractREPL, consumer::Any)
      @ REPL ~/Software/julia-1.10.0/share/julia/stdlib/v1.10/REPL/src/REPL.jl:375
   [18] (::Base.var"#1013#1015"{Bool, Bool, Bool})(REPL::Module)
      @ Base ./client.jl:432
   [19] #invokelatest#2
      @ Base ./essentials.jl:887 [inlined]
   [20] invokelatest
      @ Base ./essentials.jl:884 [inlined]
   [21] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
      @ Base ./client.jl:416
   [22] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:333
   [23] _start()
      @ Base ./client.jl:552
Test Summary:         | Error  Total  Time
ECCO fields utilities |     1      1  0.1s
ERROR: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/fpoulin/Software/ClimaOcean.jl/test/test_ecco.jl:11
```

**Comment by simone-silvestri** [comment]
It is completely safe if it is local to your system. 
Once the julia session is closed the enviromental variable specified in that way just disappears.
Do not share your password with someone else or write it down in a shared cluster.
The ECCO password is the one you would need to download the ECCO data from the NASA website, not the one you need to login. Unfortunately, these are different passwords (you can find here https://ecco.jpl.nasa.gov/drive/ what your download password is, it is your `WebDAV` password) so make sure you use that one. 
Probably we should add a clarification to the docstring.

Also it looks like the problem is really that the username is not found as an enviromental variable, which seems weird to me if you have specified it as `ECCO_USERNAME=myuser julia`. 
Maybe in the REPL check that 
```julia
ENV["ECCO_USERNAME"]
```
is indeed your username. If this does not work maybe you can export it before running julia (once you close the terminal any exported variable will disappear)

**Comment by francispoulin** [comment]
Thanks @simone-silvestri !  I did not know there were different passwords for logging into to the site and downloading the data.  

With your advice I'm happy to say that managed to download some data.

Unfortunately, it seemed to fail to find a username later on, even though it found one earlier.  


```
...
[ Info:  ... downloaded 456.1 MiB (100% complete, 1713238.134 days)
┌ Warning: The westernmost meridian of `target_grid` 0 does not coincide with the closest meridian of the bathymetry grid, -1.4210854715202004e-14.
└ @ ClimaOcean.Bathymetry ~/Software/ClimaOcean.jl/src/Bathymetry.jl:147
┌ Warning: The southernmost parallel of `target_grid` -80.0824712160682 does not coincide with the closest parallel of the bathymetry grid, -80.08333333333334.
└ @ ClimaOcean.Bathymetry ~/Software/ClimaOcean.jl/src/Bathymetry.jl:151
ERROR: LoadError: ArgumentError: Could not find the username for https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/interp_monthly/. Please provide a username in the ECCO_USERNAME environment variable.
Stacktrace:
 [1] download_dataset!(metadata::ECCOMetadata{StepRange{DateTimeProlepticGregorian, Month}, ECCO4Monthly}; url::String)
   @ ClimaOcean.DataWrangling.ECCO ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:156
 [2] download_dataset!
   @ ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_metadata.jl:144 [inlined]
 [3] ECCO_field_time_series(metadata::ECCOMetadata{…}; architecture::GPU, time_indices_in_memory::Int64, time_indexing::Oceananigans.OutputReaders.Cyclical{…}, grid::ImmersedBoundaryGrid{…})
   @ ClimaOcean.DataWrangling.ECCO ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_restoring.jl:123
 [4] ECCO_field_time_series
   @ ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_restoring.jl:112 [inlined]
 [5] ECCO_restoring_forcing(metadata::ECCOMetadata{…}; architecture::GPU, time_indices_in_memory::Int64, time_indexing::Oceananigans.OutputReaders.Cyclical{…}, mask::Field{…}, timescale::Float64, grid::ImmersedBoundaryGrid{…})
   @ ClimaOcean.DataWrangling.ECCO ~/Software/ClimaOcean.jl/src/DataWrangling/ecco_restoring.jl:271
 [6] top-level scope
   @ ~/Software/ClimaOcean.jl/prototype_omip_simulation/prototype_omip_simulation.jl:107
 [7] include(fname::String)
   @ Base.MainInclude ./client.jl:489
 [8] top-level scope
   @ REPL[4]:1
in expression starting at /home/fpoulin/Software/ClimaOcean.jl/prototype_omip_simulation/prototype_omip_simulation.jl:107
Some type information was truncated. Use `show(err)` to see complete types.
```

**Comment by glwagner** [comment]
https://github.com/CliMA/ClimaOcean.jl/blob/514135fc9a1016ec0c32081fe3914e789b3d35ca/src/DataWrangling/ecco_metadata.jl#L144-L157

Try running things one by one. I don't see how you could have set the environment variable and still got that message.

**Comment by francispoulin** [comment]
I tried it again this morning, I believe it's the exact same thing I tried yesterday, and I'm happy to say it's working.  It's still running but the first set of 11 tests passed.

It could be I did something silly or maybe the ECCO server had a problem. I have found the ECCO server can be difficult when accessing the data through another means.

Thanks again for the help and I'll close this issue, unless a problem arises before the script terminates.
